File: tests/fixtures/golden/ruby/sample.rb
require "logger"
require_relative "./helpers"

module App
  class User < Base
    attr_accessor :name

    def initialize(name)
      @name = name
    end

    def greet
      logger = Logger.new
      logger.info("greet called")
      "Hello " + @name
    end

    def self.find(id)
      User.new(id)
    end

    def sync
      find(1).update(name: @name)
      orders.map { |o| o.paid }.compact()
      helpers.list.map() { |o| o.id }
      self.notify
    end
  end
end

def greet_all(users)
  users.each { |u| puts u.greet }
end

greet_all([])

File: tests/fixtures/golden/php/sample.php
<?php
namespace App\Models;

require_once __DIR__ . "/helpers.php";
use App\Services\Logger;

interface Greeter {
    public function greet(): string;
}

trait NameAware {
    protected string $name;

    public function getName(): string {
        return $this->name;
    }
}

class User implements Greeter {
    use NameAware;

    public function __construct(string $name) {
        $this->name = $name;
    }

    public function greet(): string {
        $logger = new Logger();
        $logger->info("greet called");
        return "Hello " . $this->getName();
    }
}

function greet_all(array $users): void {
    foreach ($users as $u) {
        echo $u->greet();
    }
}

greet_all([]);
